skglm/estimators.py

```diff
@@ -229,21 +194,12 @@ class GeneralizedLinearEstimator(LinearModel):
         Number of subproblems solved to reach the specified tolerance.
     """
 
-    def __init__(self, datafit=None, penalty=None, is_classif=False, max_iter=100,
-                 max_epochs=50_000, p0=10, tol=1e-4, fit_intercept=True,
-                 warm_start=False, ws_strategy="subdiff", verbose=0):
+    def __init__(self, datafit=None, penalty=None, solver=None, is_classif=False):
         super(GeneralizedLinearEstimator, self).__init__()
         self.is_classif = is_classif
-        self.tol = tol
-        self.max_iter = max_iter
-        self.fit_intercept = fit_intercept
-        self.warm_start = warm_start
```

Comment:
warm_start would remain an attribute of GLE for me. WDYT? same for fit_intercept

Reply:
It depends where we put the boundary between what is similar with sklearn and what is not. `tol` or `max_epochs` are also attributes in sklearn yet they are abstracted away in skglm. One could argue that they are not often tuned while `fit_intercept` might be. I'd adopt the principle of least surprise, so we can expose `fit_intercept`. For `warm_start` no opinion.

Reply:
ok we need `fit_intercept` and `warm_start` for `test_check_estimator` to pass actually. debate closed.